app/controllers/posts_controller.rb

```diff
@@ -4,8 +4,28 @@ def index
     @posts = Post.where(author_id: @user.id)
   end
 
+  def all_posts
+    @user = params[:user_id]
+    @posts = Post.all
+  end
+
   def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def new
+    @author_id = params[:author_id]
```

Comment:
- [Optional] What do you think of calling `new` on the Post model in order to create a new post? Something like:

`@post = Post.new` instead of  `@author_id = params[:author_id]`